Tests/WalletConnectSignTests/WCPairingTests.swift

```diff
@@ -35,18 +35,61 @@ final class WCPairingTests: XCTestCase {
         XCTAssertEqual(pairing.expiryDate, inactiveExpiry)
     }
 
-    func testUpdateExpiry() {
+    func testUpdateExpiryForTopic() {
         var pairing = WCPairing(topic: "")
         let activeExpiry = referenceDate.advanced(by: WCPairing.timeToLiveActive)
         try? pairing.updateExpiry()
         XCTAssertEqual(pairing.expiryDate, activeExpiry)
     }
+    
+    func testUpdateExpiryForUri() {
+        var pairing = WCPairing(uri: WalletConnectURI.stub())
+        let activeExpiry = referenceDate.advanced(by: WCPairing.timeToLiveActive)
+        try? pairing.updateExpiry()
+        XCTAssertEqual(pairing.expiryDate, activeExpiry)
+    }
 
-    func testActivate() {
+    func testActivateTopic() {
         var pairing = WCPairing(topic: "")
         let activeExpiry = referenceDate.advanced(by: WCPairing.timeToLiveActive)
+        XCTAssertFalse(pairing.active)
+        pairing.activate()
+        XCTAssertTrue(pairing.active)
+        XCTAssertEqual(pairing.expiryDate, activeExpiry)
+    }
+    
+    func testActivateURI() {
+        var pairing = WCPairing(uri: WalletConnectURI.stub())
+        let activeExpiry = referenceDate.advanced(by: WCPairing.timeToLiveActive)
+        XCTAssertFalse(pairing.active)
         pairing.activate()
         XCTAssertTrue(pairing.active)
         XCTAssertEqual(pairing.expiryDate, activeExpiry)
     }
+    
+    func testUpdateExpiry_WhenValueIsGreaterThanMax_ShouldThrowInvalidUpdateExpiryValue() {
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: referenceDate)
+        XCTAssertThrowsError(try pairing.updateExpiry(40 * .day)) { error in
+            XCTAssertEqual(error as! WCPairing.Errors, WCPairing.Errors.invalidUpdateExpiryValue)
+        }
+    }
+    
+    func testUpdateExpiry_WhenNewExpiryDateIsLessThanExpiryDate_ShouldThrowInvalidUpdateExpiryValue() {
+        let expiryDate = referenceDate.advanced(by: 10 * .minute)
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: expiryDate)
+        XCTAssertThrowsError(try pairing.updateExpiry(40 * .day)) { error in
+            XCTAssertEqual(error as! WCPairing.Errors, WCPairing.Errors.invalidUpdateExpiryValue)
+        }
+    }
+
+    func testActivate_WhenUpdateExpiryFails_ShouldActivateAndThrowError() {
+        let expiryDate = referenceDate.advanced(by: 10 * .minute)
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: expiryDate)
+        XCTAssertFalse(pairing.active)
+        pairing.activate()
+        XCTAssertThrowsError(try pairing.updateExpiry(10 * .minute)) { error in
```

Comment:
`activate()` makes a call to `updateExpiry()` internally (which sets a 30-day expiry). 

Is there a reason to call activate before a throwing update? Does it make a difference for this test to invert the order?

Reply:
I was trying to check the error inside the `activate()` function, but if I understood it right the error dies inside `activate()`. I changed the test, now checking for the `expiryDate` update. If an error occurs, it shouldn't be updated.